src/funnels/Swapper.sol

```diff
@@ -71,21 +72,34 @@ contract Swapper {
         emit Deny(usr);
     }
 
-    function setLimits(address src, address dst, uint64 hop, uint128 cap) external auth {
+    function setLimits(address src, address dst, uint128 cap, uint64 hop) external auth {
+        uint128 amt = limits[src][dst].amt;
         limits[src][dst] = PairLimit({
-            hop:  hop,
-            zzz:  limits[src][dst].zzz,
-            cap: cap
+            cap: cap,
+            hop: hop,
+            amt: amt > cap ? cap : amt,
+            zzz: limits[src][dst].zzz
         });
-        emit SetLimits(src, dst, hop, cap);
+        emit SetLimits(src, dst, cap, hop);
     }
 
     function swap(address src, address dst, uint256 amt, uint256 minOut, address callee, bytes calldata data) external auth returns (uint256 out) {
         PairLimit memory limit = limits[src][dst];
-        require(block.timestamp >= limit.zzz + limit.hop, "Swapper/too-soon");
-        limits[src][dst].zzz = uint64(block.timestamp);
 
-        require(amt <= limit.cap, "Swapper/exceeds-max-amt");
+        unchecked {
```

Comment:
Not a huge save but note that moving memory to stack  saves gas for variables that are read multiple times or written to temporarily.

In this case you can consider using something like:
```
uint64 _zzz = limit.zzz;
uint128 _amt = limit.amt;
```

Reply:
ok